### ISSUE TITLE
Kokkos does not export AMD flags to Trilinos

### ISSUE THREAD

**Issue by lucbv** [kokkos/kokkos#5505]
**Describe the bug**
When configuring Trilinos to build Kokkos with HIP enabled the `--offload-arch=gfx90N` flag is apparently not added to Trilinos' flags.
This leads to obvious compilation issues, setting the environment variable `ROCM_TARGET_LST` fixes the issue but defeats the point of specifying the ARCH to Kokkos at configure time.
Adding the following logic in CMakeList.txt at line 239 also fixes this issue:
```
  IF(KOKKOS_ENABLE_HIP)
    FOREACH(OPTION ${KOKKOS_AMDGPU_OPTIONS})
      STRING(FIND "${OPTION}" " " OPTION_HAS_WHITESPACE)
      IF(OPTION_HAS_WHITESPACE EQUAL -1)
        LIST(APPEND KOKKOS_COMPILE_OPTIONS_TMP "${OPTION}")
      ELSE()
        LIST(APPEND KOKKOS_COMPILE_OPTION_TMP "\"${OPTION}\"")
      ENDIF()
    ENDFOREACH()
  ENDIF()
```

**Comment by jjellio** [comment]
Seems like this would get fixed if kokkos eliminated it's special case of HIPCC (which cmake hates anyway). #5453 
I'm guessing if you used the AMD compiler directly, Kokkos/Cmake would find ROCm (another huge headache) - but it would probably propagate the flags. (I haven't tested it)

**Comment by lucbv** [comment]
In my case above the underlying issue is that I am compiling on a head node without GPUs where `rocm_agent_enumerator` does not find the target architecture. If I build on a compute node there no problem anymore.
The main issue is that Kokkos does not appear to tell Trilinos which target should be used based on the Kokkos_ARCH. My CMake attempt above works fine but is probably not the exact fix that Kokkos wants to implement?

**Comment by jjellio** [comment]
It seems, if we didn't use `hipcc` and Kokkos let CMake do its thing, then wouldn't cmake find/propagate the information? RIght now, kokkos isn't really enabling "hip" via CMake, because HIPCC obscures that by setting everything itself.  I've been manually setting all my flags (and using or not using hipcc) - but I think since we use CMake, Cmake is at odds with `hipcc` as a compiler. I can't do a Kokkos build w/Tribits (my issue) - so we can't really eval if Cmake would find and propagate the information correctly. But Bruno seems to show that Kokkos w/pure cmake and non-hipcc does set all that stuff (which means it should be propagating)

**Comment by crtrott** [comment]
Note: if you want to let CMake do its thing, then Trilinos and all downstream packages would need to commit to using the cmake language mechanism. And that means THEY need to mark all their source files as "HIP", "CUDA" etc. files. We do support that at least with CUDA, but it is fairly clunky. The problem is that you can't "inherit" a language from a target dependency.

**Comment by masterleinad** [comment]
This doesn't appear to be a problem for `SYCL`. `SYCL` (and `OpenMPTarget`) use `COMPILER_SPECIFIC_FLAGS` directly and I can see the architecture flags when compiling `Trilinos`.

**Comment by PhilMiller** [comment]
Fixed in 3.7.01 with #5571
